packages/grafana-data/src/types/pluginExtensions.ts

```diff
@@ -167,6 +167,13 @@ export type PluginExtensionEventHelpers<Context extends object = object> = {
   context?: Readonly<Context>;
   // Opens a modal dialog and renders the provided React component inside it
   openModal: (options: PluginExtensionOpenModalOptions) => void;
+  /**
+   * @internal
+   * Opens the extension sidebar with the registered component.
+   * @param componentTitle The title of the component to be opened in the sidebar.
+   * @param props The props to be passed to the component.
+   */
+  openSidebar?: (componentTitle: string, props?: any) => void;
```

Comment:
I don't think openSidebar is optional? Isn't it always available like openModal. 🤔 

Can we also avoid the usage of `any` please?

nit: I'd remove the `@internal`. As this is part of the public api, marking it as internal doesn't really change the fact and will likely confuse plugin devs about its usage.

Reply:
You are right about the optional - will fix.

Regarding `@internal`:
We scoped the extension sidebar deliberately to a few plugins:
https://github.com/grafana/grafana/blob/main/public/app/core/components/AppChrome/ExtensionSidebar/ExtensionSidebarProvider.tsx#L13-L17

We even changed the icon to be very specific for these apps: https://raintank-corp.slack.com/archives/CKA461W3X/p1744269730074289

Is there any better way to provide this API to only those other plugins?

Reply:
Ah, went with `Record<string, unknown>` in 887f7527c14f192b1c8afb41f10af9c65c1016aa

Reply:
>Is there any better way to provide this API to only those other plugins?

I think it makes sense to have it as part of the helpers (as we have for openModal) so in this case I'd say there isn't really a better way.

As for `@internal` it's fair enough if you wanna use it, I just find it weird as it doesn't prevent a developer from using it and we don't error or warn if folk do with something like eslint.

Reply:
Yea, right. I'd rather keep it, as it gives some intention that this API is not stable enough to rely on yet.